hack/generator/pkg/astmodel/reference_graph.go

```diff
@@ -39,17 +39,21 @@ func CollectArmSpecAndStatusDefinitions(definitions Types) TypeNameSet {
 		armName := CreateArmTypeName(name)
 
 		if _, ok = definitions[armName]; !ok {
-			return TypeName{}, errors.Errorf("Couldn't ARM type find %q", armName)
+			return TypeName{}, errors.Errorf("Couldn't find ARM type %q", armName)
 		}
 
 		return armName, nil
 	}
 
-	// TODO: We should be using a better way to identify ARM types. I believe
-	// TODO Bevan is working on it.
 	armSpecAndStatus := make(TypeNameSet)
 	for _, def := range definitions {
-		if resourceType, ok := def.Type().(*ResourceType); ok {
+
+		if IsStoragePackageReference(def.Name().PackageReference) {
```

Comment:
As mentioned in a different comment I think you could technically remove this and things would still work (because `ResolveResourceType` won't resolve `StorageTypes`?

If you did do that you could also delete this method.

I'm fine with it this way too though if you prefer so feel free to just resolve this if you wanna leave it as is

Reply:
Comments added - let me know whether they're sufficiently clear

Reply:
Assuming the comments are sufficiently clear.